### PR TITLE
Feature: Playback position

### LINKED ISSUE #84: Feature: Track Progress Bar
Been having a think about how we can achieve this easily. 

The Player can now tick over the track every second and update a Redis key `fm:player:progress` to hold the `%` value we are through the track. This can be returned as part of the `/player/current` resource allowing you to place an element at the correct position of the progress bar. This will not update if the player is in a paused state.

When you get the `play` event you can start your progress bar. How fast the bar moves will need to be calculated based on the duration of the track, so the bar should move slower for a 5min track vs a 3min track. When you get a `pause` event you can stop the progress bar. When you get a `resume` event you can start the bar off again.

This should give us everything we need to display track progress.

In the future we can also support Seeking, this will mean a new event, probably called `seek_to` and just send a `%` value to seek to in the track, like `23%`.

Should work?

### PR DESCRIPTION
**Pull request by jamesjwarren**
# Feature: Playback position
This PR adds tracking the elapsed time for currently playing track with progress bar. Closes #84 

screenshot 2015-06-16 17 19 54

### CODE REVIEW

**Comment by krak3n** [comment]
OOooooooo

:boom: :exclamation:

**Comment by peeklondon** [comment]
WOW! :star2: :high_brightness:

File: app/js/player/controllers/PlayerCtrl.js [hunk 1]
```diff
@@ -191,6 +205,10 @@ angular.module("FM.player.PlayerCtrl",[
         $scope.$on("fm:player:setVolume", $scope.onSetVolume);
 
         $scope.getAllData();
+
+        $scope.$on("$destroy", function(){
+            $scope.trackPositionTimer.stop();
+        });
```
Comment (bcmrbvm) [inline app/js/player/controllers/PlayerCtrl.js]: Not major but here you could just do:
```javascript
$scope.$on("$destroy", $scope.trackPositionTimer.stop);
```

File: app/js/player/services/TrackTimer.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+         * @private
+         */
+        var startTime = 0;
+
+        /**
+         * Time on the clock when last stopped in milliseconds
+         * @property {Number} lastStopTime
+         * @private
+         */
+        var	lastStopTime = 0;
```
Comment (bcmrbvm) [inline app/js/player/services/TrackTimer.js]: spacing between `var` and the variable name

File: app/js/player/services/TrackTimer.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+         * @method getTime
+         * @returns {Number} total elapsed time
+         * @private
+         */
+        var getTime = function getTime () {
+            return lastStopTime + (now() - startTime);
+        };
+
+        /**
+         *
```
Comment (bcmrbvm) [inline app/js/player/services/TrackTimer.js]: unneeded whitespace

**Comment by bcmrbvm** [comment]
:+1:

**Pull request merged**